### DISCUSSION TITLE
setConcurrentConnections config is not working as multi-threading

### DISCUSSION THREAD

**Discussion by lantian1989** [relayrides/pushy#979] [Q&A]
Hi there,

We are trying to use concurrent connections for APNS clients by setting ``setConcurrentConnections`` with configurable values, for example, like 4 or 8. However, from our logging and metrics system, we dont see our APNS clients are working as multi-threading, it is still single thread. Does anyone have the same experience and how could we actually set it as concurrent connection? Thanks!

**Comment by jchambers** [comment]
Please see the ["threads, concurrent connections, and performance" section of the "best practices" wiki page](https://github.com/jchambers/pushy/wiki/Best-practices#threads-concurrent-connections-and-performance). If you want more threads, you need to expand your [event loop group](https://pushy-apns.org/apidocs/latest/com/eatthepath/pushy/apns/ApnsClientBuilder.html#setEventLoopGroup(io.netty.channel.EventLoopGroup)).
